src/main/java/com/limechain/runtime/version/ApiVersions.java

```diff
@@ -0,0 +1,66 @@
+package com.limechain.runtime.version;
+
+import com.limechain.runtime.version.scale.ApiVersionReader;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.scale.ScaleReader;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+public class ApiVersions {
+    private final List<ApiVersion> entries;
+
+    private ApiVersions(List<ApiVersion> entries) {
+        this.entries = entries;
+    }
+
+    /**
+     * Constructs a new ApiVersions from an unmodifiable view of the passed list
+     */
+    public static ApiVersions of(List<ApiVersion> apiVersions) {
+        return new ApiVersions(Collections.unmodifiableList(apiVersions));
+    }
+
+    public static ApiVersions decodeNoLength(byte[] scaleEncoded) {
+        ScaleCodecReader reader = new ScaleCodecReader(scaleEncoded);
+        ScaleReader<ApiVersion> apiVersionReader = new ApiVersionReader();
+        List<ApiVersion> apiVersionsRaw = new LinkedList<>();
+
+        //TODO: Think about checking if the length is divisible by the fixed byte size of an encoded api version
+        while (reader.hasNext()) {
+            try {
+                apiVersionsRaw.add(reader.read(apiVersionReader));
+            } catch (RuntimeException e) {
+                break; // Presume end of iteration (all is ok), although other exceptions are possible
+            }
```

Comment:
It's probably good to have a log here, if the exception is expected the log could be set to a lower logging level, but it's good to have some info about the exception.

Reply:
Actually, an exception was not expected per se, it's rather a really exceptional scenario, it was a miss on my part.
Take a look again if we're satisfied with this handling (javadoc improved as well): a037d61